core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -170,6 +183,137 @@ export class ContributionAndReviewService {
+                fetchSuggestionsResponse.
+                target_id_to_opportunity_dict
+                [suggestion.target_id])
+            };
+            responseSuggestionIdToDetails[
+              suggestion.suggestion_id] = suggestionDetails;
+          });
+          return {
+            suggestionIdToDetails: responseSuggestionIdToDetails,
+            more: false
+          };
+        })
+    );
+  }
+
+  // Function to sort translation cards by state.
+  sortTranslationSuggestionsByState(
+      translationSuggestions: SuggestionBackendDict[],
+      states: States,
+      initStateName: string | null
+  ): SuggestionBackendDict[] {
+    // Obtain the state names in the order of content flow in the lesson.
+    if (!initStateName) {
+      return translationSuggestions;
+    }
+
+    const stateNamesInOrder = this.computeGraphService.
+      computeBfsTraversalOfStates(
+        initStateName,
+        states,
+        initStateName
+      );
+    const translationSuggestionsByState = (
+      ContributionAndReviewService
+      .groupTranslationSuggestionsByState(translationSuggestions));
+    const sortedTranslationCards: SuggestionBackendDict[] = [];
+
+    for (const stateName of stateNamesInOrder) {
+      const cardsForState = (
+        translationSuggestionsByState.get(stateName) || []);
+      cardsForState.sort(ContributionAndReviewService
+        .compareTranslationSuggestions.bind(this));
+      sortedTranslationCards.push(...cardsForState);
+    }
+    return sortedTranslationCards;
+  }
+
+  static groupTranslationSuggestionsByState(
+    translationSuggestions: SuggestionBackendDict[])
+    : Map<string, SuggestionBackendDict[]> {
+    const translationSuggestionsByState = new Map<
+    string, SuggestionBackendDict[]>();
+
+    for (const translationSuggestion of translationSuggestions) {
+      const stateName = translationSuggestion.change.state_name;
+      const suggestionsForState = translationSuggestionsByState.get(stateName) || [];
+      suggestionsForState.push(translationSuggestion);
+      translationSuggestionsByState.set(stateName, suggestionsForState);
+    }
+    return translationSuggestionsByState;
+  }
+
+  // Returns the type order for a given content ID.
+  static getTypeOrder(contentId: string): number {
+    const type = contentId.split('_')[0];
+    const order: { [key: string]: number } = {
+      'content': 0, // eslint-disable-line quote-props
+      'interaction': 1, // eslint-disable-line quote-props
+      'feedback': 2, // eslint-disable-line quote-props
+      'default': 3, 
+      'hints': 4, // eslint-disable-line quote-props
+      'solution': 5 // eslint-disable-line quote-props
+    };
+    return order.hasOwnProperty(type) ? order[type] : Number.MAX_SAFE_INTEGER;
+  }
+
+  // Returns index for a given content ID.
+  static getIndex(contentId: string): number {
+    const index = parseInt(contentId.split('_')[1]);
+    return isNaN(index) ? Number.MAX_SAFE_INTEGER : index;
+  }
```

Comment:
Move these below `compareTranslationSuggestions`. You generally want to order your class in order of usage.

Reply:
Ah alright thanks, Done.